lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp

```diff
@@ -78,22 +83,13 @@ UsdUndoInsertChildCommand::UsdUndoInsertChildCommand(const UsdSceneItem::Ptr& pa
     }
     _usdDstPath = parent->prim().GetPath().AppendChild(TfToken(childName));
 
-    _childLayer = MayaUsdUtils::defPrimSpecLayer(childPrim);
-    if (!_childLayer) {
-        std::string err = TfStringPrintf("No child prim found at %s", childPrim.GetPath().GetString().c_str());
-        throw std::runtime_error(err.c_str());
-    }
+    _childLayer = _stage->GetEditTarget().GetLayer();
 
     // If parent prim is the pseudo-root, no def primSpec will be found, so
     // just use the edit target layer.
-    _parentLayer = parentPrim.IsPseudoRoot() ?
-        _stage->GetEditTarget().GetLayer() :
-        MayaUsdUtils::defPrimSpecLayer(parentPrim);
-
-    if (!_parentLayer) {
-        std::string err = TfStringPrintf("No parent prim found at %s", parentPrim.GetPath().GetString().c_str());
-        throw std::runtime_error(err.c_str());
-    }
+    _parentLayer = parentPrim.IsPseudoRoot() 
+        ? _stage->GetEditTarget().GetLayer() 
+        : MayaUsdUtils::defPrimSpecLayer(parentPrim);
 }
```

Comment:
Both "_childLayer" and "_parentLayer" should be now in the correct target layers, therefore they can simply take "_stage->GetEditTarget().GetLayer()".

@ppt-adsk does "_parentLayer"  still need to check for "parentPrim.IsPseudoRoot()"? I would like to understand when this happens.